autoload/go/template.vim

```diff
@@ -9,6 +9,7 @@ function! go#template#create() abort
   let l:root_dir = fnamemodify(s:current_file, ':h:h:h')
 
   let l:package_name = go#tool#PackageName()
+  let l:filename = expand('%:t')
```

Comment:
There's no need to move this; it's only used; keep its declaration close to where it's used aids in readability.